Clarify Steering Council roles (#23)
neoforged/governance: core.md

#### Steering Council

The Steering Council helps to ensure good communication within teams by making sure significant work, PRs, rule changes, or decision are backed by a team's vote that is outlined in [/voting.md](/voting.md). The result of that team's vote will be then enforced, whether it be the Maintainer team, Moderator team, or any other. While any team member can put up votes themselves, the Steering Council acts as a dedicated monitoring position to watch for anything that could slip by without a vote.

> In other words, the members of the Steering Council does not give out orders or demands. They cannot tell people what to do. Instead, if something comes up and a majority of the Steering Council members believes it is significant, they can say that the thing needs a decision from the whole team and put up a vote for that team. This helps teams be aware of what others members on their team are doing and helps reduce people being blindsided by big changes. It also addresses the problem of a single member making sweeping changes under other member's noses.

Reviewer: The wording of the first part of this paragraph should be improved a little: "something", "thing" should be clarified.

Regarding the second part: the "blindsided" and "single member making sweeping changes" parts do not belong here, and I would just remove them.

Author: Oh you must had been reviewing before I just pushed up the new wording. The something and thing was replaced moment ago

Author: resolving this as it was already fixed

> The Steering Council has at all times an odd number of active members, such that stalemates are not possible when deciding if something warrants a vote or not. The members of the Steering Council are decided by an annual internal vote to ensure fairness and cycling of council members over time. Details on the Steering Council voting process can be found at [/steering-council-voting.md](/steering-council-voting.md).

Reviewer: This `steering-council-voting.md` document doesn't exist yet. 🤔

Author: It will once Maty's PR is merged. He took the existing voting.md doc and renamed it to steering-council-voting.md which my PR will be linking to

Author: Maty PR is merged now

> A member may choose to become inactive and appoint another Project team member in their place until they return. If a Steering Council member cannot be reached and there is an important time-sensitive Steering Council vote, an emergency Steering Council election will be held in accordance with [/steering-council-voting.md](/steering-council-voting.md). The newly elected member(s) will now act as the Steering Council member in question until their return or their term expires.

Reviewer: Is this new?

Author: The old wording was `A member may choose to become inactive and appoint another in their place until they return.` which was vague and didn't account for if a member disappears without appointing a new member. There was talk in internal private which resulted in the general idea of this new paragraph to be more encompassing and handles edge cases better that the previous form did not cover.

Author: Gonna close this as you were just asking if it was new or not and I answered

#### Steering Council

> The Steering Council ensures solid team communication by making sure significant work, PRs, rule changes, or decision are backed by a team's vote that is outlined in [/voting.md](/voting.md). The result of that team's vote will be then enforced, whether it be the Maintainer team, Moderator team, or any other. While any team member can call for team votes, the Steering Council acts as a dedicated monitoring position to watch for anything that could slip by without a vote.

Reviewer 2: ```suggestion
The Steering Council ensures that members of a Team are in agreement regarding topics under their purview. To that end, the Steering Council monitors in-progress affairs, reserving the right to trigger votes.
```

Reviewer: "reserving the right" is a bit weird... anyone can hold a vote so this might be misinterpreted.

Reviewer 2: Maybe "having the right"? Not sure, though the voting document already states that anyone and SC members can trigger a vote so this might be redundant?

Author: Errr, I rather not remove the link to the voting doc. The whole point of Steering Council now was to help make sure voting doc is followed and votes are done when they should be done

Author: alright I see, you moved it to the other suggested commit

Author: tried new wording, will close this convo

> In other words, the members of the Steering Council does not give out orders or demands. They cannot give orders to team members of what to do. Instead, if a decision or action comes up and a majority of the Steering Council members believes it is significant, they can say that the situation needs to be communicated to the relevant team and put up a vote for that team.

Reviewer 2: ```suggestion
In other words, when the Council is made aware of a situation, and if it decides (after a vote between the members of the Council) that the situation warrants the involvement of an entire Team, they can trigger a vote as described by the [voting rules](./voting.md). The Council cannot influence the vote, and its sole responsibility is to enforce the outcome.
```

Reviewer: The proposed last sentence is a bit confusing IMO: The SC cannot influence the vote, but its members are free to vote (representing themselves, not the SC) if they are part of the relevant teams.

Reviewer 2: Yeah my intent was to point at the fact that it does not tie break nor is the council able to vote on the matter itself, just on.. whether there should be a vote. Not sure how else to word it, hmm

Author: tried new wording, will close this convo

> The Steering Council has an odd number of active members at all times, such that stalemates are not possible when deciding if something warrants a vote or not. The members of the Steering Council are decided by an annual internal vote to ensure fairness and cycling of council members over time. Details on the Steering Council voting process can be found at [/steering-council-voting.md](/steering-council-voting.md).

Reviewer 2: I don't see a reason to change this. The previous wording was fine and there is no reason to specify exactly when a stalemate is possible.

> A member may choose to become inactive and appoint another Project team member in their place until they return, or their term expires, whichever is sooner. If a Steering Council member cannot be reached and a replacement is desired, a Steering Council election will be held in accordance with [/steering-council-elections.md](/steering-council-elections.md). The newly elected member(s) will now act as the Steering Council member in question. The original uncontactable Steering Council member(s) effectively lost their seat(s) and only gets it back if they win the next normal Steering Council elections.

Reviewer 2: ```suggestion
A member may choose to become inactive and appoint another Project team member in their place until they return, or their term expires, whichever is sooner. If a Steering Council member cannot be reached and a replacement is desired, a Steering Council election will be held in accordance with [/steering-council-elections.md](/steering-council-elections.md). The newly elected member(s) will now replace the Steering Council member in question until the next yearly election.
```

> A member may choose to become inactive and appoint another Project team member in their place until they return. If a Steering Council member cannot be reached and there is an important time-sensitive Steering Council vote, an emergency Steering Council election will be held in accordance with [/steering-council-voting.md](/steering-council-voting.md). The newly elected member(s) will now act as the Steering Council member in question until the original Steering Council member(s) return or their term expires.

Reviewer 2: Please describe the emergency election process within that document.
I propose that remaining SC members take up the job of election officer, accept **only** self nominations for 24 hours then put up the vote for 24 hours.

Reviewer: Given the new SC role, couldn't the rest of the SC simply continue to operate (with an even number of people)? Taking 48h to resolve an "emergency" is not reasonable.

Author: I added entries to the steering council voting doc. ~~But what exactly is an election officer? Do we have that defined anywhere?~~ Election officer is described in steering-council-elections.md doc

Author: It does seem like it would be hard for there to be any important time-sensitive decision from steering council now. Emergency vote could be reserved for when there's only 1 active Steering council (the other two confirmed to be long-term inactive without replacement or left)

Author: emergency election part has been removed now

A member may choose to become inactive and appoint another in their place until they return.
A member may choose to become inactive and appoint another Project team member in their place until they return. If a Steering Council member cannot be reached and there is an important time-sensitive Steering Council vote, an emergency Steering Council election will be held in accordance with [/steering-council-voting.md](/steering-council-voting.md). The newly elected member(s) will now act as the Steering Council member in question until the original Steering Council member(s) return or their term expires.

At least one member of the Council is promoted from other teams via an annual internal vote, to ensure that the Council always has fresh insight from a new team member with a different view on situations, and thus does not stagnate over time.

> Once a year, an internal voting process inspired by [the process implemented in the Python ecosystem](https://peps.python.org/pep-0013/) will determine the members that will be promoted to the Steering Council via the [Helios voting system](https://heliosvoting.org/). All three members of the Council will be elected, at least one of which must be a member that has not been on the Council in the previous term.

Reviewer 2: Removing this is not within the scope of this PR (that is, clarifying the role of the Council). Please add it back and move the "details can be found" section from above to the last paragraph here.

Author: We decided to move these lines to the steering-council-elections.md
 file. Basically this line was describing the voting process which does not belong in this core.md file. The process and reference for voting should stay in steering-council-elections.md document

Reviewer 2: Ah yes indeed, I didn't notice the
> At least one new member should be elected to the Steering Council in each year, to get fresh views and opinions into the council.

from the voting document

#### Steering Council

> The Steering Council are the representatives of the Project, final check for if a Project member is to be demoted or banned, and tiebreakers for perfectly split votes.

Reviewer 2: ```suggestion
The Steering Council is the representative of the Project in the community. It has the power to cast tiebreaking votes in Team votes with a perfectly split outcome, and it also renders a final decisions in cases involving the demotion or removal from the community of Project members. 
```
The use of is here is correct as it refers to the council as an institution, and not its members (see the original governance pr).

Author: closing comments as Steering Council was completely redone

> The Council has at all times an odd number of active members, such that stalemates are not possible when making decisions. The members of the Council are decided by an annual internal vote to ensure fairness and cycling of council members over time. Details on the Steering Council voting process can be found at [/steering-council-voting.md](/steering-council-voting.md).

Reviewer 2: For now I would keep the part about one new SC must be elected. We can discuss the whole topic in a separate PR that also outlines term limits (probably the upcoming one about elections in general)

> A member may choose to become inactive and appoint another Project team member in their place until they return. If a Steering Council member is not able to be reached and there is an important time-sensitive Steering Council vote that needs to be performed, an emergency Steering Council election will be held in accordance with [/steering-council-voting.md](/steering-council-voting.md). The newly elected member(s) will now act as the Council member in question until their return or their term expires.

Reviewer 2: ```suggestion
A member may choose to become inactive and appoint another Project team member in their place until they return. If a Steering Council member cannot be reached and there is an important time-sensitive Steering Council vote pending, an emergency Steering Council election will be held in accordance with [/steering-council-voting.md](/steering-council-voting.md). The newly elected member(s) will now act as the Council member in question until their return or their term expires.
```

> The Council will act as representatives for the Project to the community and to any external entity. They will essentially act as a figurehead and a middleman to facilitate communication. 

Reviewer 2: As above, the council is a singular entity. You're also using act twice here. What about be?
